src/services/EdFi.DataManagementService.Api.Tests.E2E/Features/SchoolYearValidation.feature

```diff
@@ -0,0 +1,208 @@
+@SchoolYear
+Feature: School Year Reference Validation
+  Reference validation on School Years
+
+  """
+  @StudentEducationOrganizationAssociation Hook that will create the following data for the required tests
+  createStudent(studentUniqueId)
+  createSchool(schoolId)
+  createCourse(courseCode, schoolId)
+  createSession(schoolId, schoolYear, sessionName)
+  createCourseOffering(localCourseCode, courseCode, schoolId, schoolYear, sessionName)
+  createSection(localCourseCode, schoolId, schoolYear, sessionName, sectionIdentifier)
+  """
+
+  @Ignore @StudentEducationOrganizationAssociation
+  Scenario: Post valid request using a existing year and make sure response is correctly
+    Given a existing school year
+    When sending a POST request to "/ed-fi/studentSectionAssociation" with body
+    """
+        {
+        "beginDate": "2024-04-25",
+        "sectionReference": {
+            "localCourseCode": "string",
+            "schoolId": 0,
+            "schoolYear": 0,
+            "sectionIdentifier": "string",
+            "sessionName": "string",
+        },
+        "studentReference": {
+            "studentUniqueId": "string",
+        }
+        }
+    """
+    Then the response code is 200
+    And the response body is
+    """
+    """
+
+  @Ignore @StudentEducationOrganizationAssociation
+  Scenario: Post invalid request using a non existing year and make sure to receive an appropiate error message
+    When sending a POST request to "/ed-fi/studentSectionAssociation" with body
+    """
+        {
+        "beginDate": "2024-04-25",
+        "sectionReference": {
+            "localCourseCode": "string",
+            "schoolId": 0,
+            "schoolYear": 0,
+            "sectionIdentifier": "string",
+            "sessionName": "string",
+        },
+        "studentReference": {
+            "studentUniqueId": "string",
+        }
+        }
+    """
+    Then the response code is 404
```

Comment:
Should be 400 - because the URL in line 41 _does_ exist, but the request body has a mistake in it (a bad school year).